### ISSUE TITLE
Prevent files from being downloaded again  

### ISSUE THREAD

**Issue by itsyigid** [hruzgar/ukey-downloader#4]
Running the script multiple times will download every available file in UKEY regardless of the fact that they've already been downloaded. This consumes time and bandwidth in many cases (students in dormitories are known to have limited quotas for download)

We can prevent this from happening by keeping a file (a local database) of every file that the script completely downloaded. I offer the following method:

Files uploaded to UKEY have uniquely different names on the second column of the table where they're kept. We can utilize a hash function (a relatively simple one like MD5 or SHA1 is enough) to produce the [digest](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of **file "name"** and store this ciphertext in the local database. **File "name"** here is not necessarily same as the name of file that we will download.

When the script is executed, it will load all hashes stored in the local database, (remember that these hashes are for completed files)
and start navigating through UKEY whilst comparing hash of names on the table against the local database. Any "new name" which its hash is not in the local database, will then be downloaded.

**Comment by hruzgar** [comment]
Yeah I also had this issue in my mind when first creating the project. But hashing the filenames didn't came to my mind. Jeez. My first thought was to check if the filename already exists (checking all the files in the subfolder). But your method of doing this is much more convenient. We also should cut the database into the different classes to increase the performance (time complexity etc.).

**Comment by itsyigid** [comment]
Cutting database into classes can be implemented after we tackle this issue. Currently I am working on this and plan to release it with v1.5 tag
